src/main/java/org/zalando/nakadi/domain/BatchItem.java

```diff
@@ -58,4 +59,8 @@ public void updateStatusAndDetail(final EventPublishingStatus publishingStatus,
         response.setPublishingStatus(publishingStatus);
         response.setDetail(detail);
     }
+
+    public int getEventSize() {
+        return event.toString().getBytes(StandardCharsets.UTF_8).length;
```

Comment:
@antban I agree that calling `toString` might have an impact in performance, so I suggest to try load test on staging. But in terms of behaviour from the producer's point of view, we better provide a clear contract. If we had a variable contract, sometimes accepting 999,990 bytes, sometimes 999,985 bytes, depending on enrichment and so on, users might get frustrated by the lack of consistency.

For example, we are going to add schema version which enlarges the metadata even further. It would possibly break publishers with event close to the limit.

How we are going to achieve this is another problem.